src/dashboard/pages/index/controller.py

```diff
@@ -1,10 +1,110 @@
 """Index controller module.
 
-This module is intented to contain callbacks used in the index page. As
-of now it is just a stub.
+Callbacks to the index page.
+Callbacks to toggle add dashboard modal
+and submitting a new dashboard.
 """
 
+from dash import Input, Output, State, callback, ctx
 
-def do_stuff() -> bool:
-    """Stub function."""
+from dashboard.models.user import add_dashboard
+
+input_css = "p-3 rounded-md shadow-inner bg-background "
+
+
+@callback(
+    Output({"type": "modal-dialog", "id": "add-dashboard-dialog"}, "open", allow_duplicate=True),
+    Input("create-dashboard-btn", "n_clicks"),
+    Input("create-dashboard-btn-carousel", "n_clicks"),
+    Input("cancel-btn", "n_clicks"),
+    prevent_initial_call=True,
+)
+def toggle_create_dashboard(create_btn: int, create_btn_2: int, cancel_btn: int) -> bool:
+    """Toggles the create dashboard modal.
+
+    Args:
+        create_btn (int): create dashboard button main
+        create_btn_2 (int): create dashboard button carousel
+        cancel_btn (int): cancel button clicks
+
+    Returns:
+        bool: value to toggle modal
+    """
+    if "cancel-btn" == ctx.triggered_id:
+        return False
     return True
```

Comment:
This can be simplified. Either `return ctx.triggered_id != "cancel-btn"` or split into two callbacks. I think the second solution would be clearer.